cmd/kubeadm/app/util/version.go

```diff
@@ -95,6 +94,13 @@ func KubernetesReleaseVersion(version string) (string, error) {
 			klog.Infof("falling back to the local client version: %s", clientVersion)
 			return KubernetesReleaseVersion(clientVersion)
 		}
+
+		if clientVersionErr != nil {
+			// TODO: theoretically clinetVersion is always accessible, but only if build system is tweaked correctly
+			klog.Warningf("could not obtain client version; using remote version: %s", body)
```

Comment:
Very true! I guess if both versions are unavailable we can do nothing, but `os.Exit(1)` with some explanation. I'll squash these 3 commits and submit one more PR.

Reply:
@rojkov FYI, @bart0sh is currently working on a PR (?) that may add a constant that refers to the actual current version of the client (even if not obtained from a build system).

this could be our final fallback.
let's decide if we want to do that.

Reply:
Yeap, I'll discuss with Ed what would be the best option.